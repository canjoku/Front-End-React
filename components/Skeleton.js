import { Component } from 'react'
import PropTypes from 'prop-types'

class Skeleton extends Component {

  preLoader () {
    // this function initialises some pre loaders for the Skeleton. I have set it to a default of 4
    let preLoaders = []

    // default background style for each preloader
    let elementStyle = {
      'background': 'linear-gradient(to right, '+ this.props.color +', #f7f7f7)'
    }

    // the function performs an iteration to return 3 pre loaders for the skeleton
    for (let index = 0; index < 3; index++) {
      preLoaders.push(
        <div key={index} className="s-skeleton__element" style={elementStyle}></div>
      )
    }

    return preLoaders
  }
  // this function provides loader rows. the number of required rows can be provided via props
  loaderRows () {
    const rows = [];
    for (let index = 0; index < this.props.count; index++) {
      rows.push(
        <div key={index} className="s-skeleton">
          {this.preLoader()}
        </div>
      )
    }
    return rows
  }

  render () {
    return (
      <div>
        {this.loaderRows()}
      </div>
    )
  }

  static propTypes = {
    color: PropTypes.string,
    count: PropTypes.number
  }

  static defaultProps = {
    count: 1,
    color: '#173F5F'
  }

}

export default Skeleton
