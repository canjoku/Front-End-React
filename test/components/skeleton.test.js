import Skeleton from '../../components/Skeleton'

test('<Skeleton /> renders correctly', () => {
  const component = shallow(<Skeleton count={2} color="#F6D55C" />)
  expect(component).toMatchSnapshot()
})
