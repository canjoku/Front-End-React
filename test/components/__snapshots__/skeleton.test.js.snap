// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Skeleton /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Skeleton
    color="#F6D55C"
    count={2}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="s-skeleton"
        >
          <div
            className="s-skeleton__element"
            style={
              Object {
                "background": "linear-gradient(to right, #F6D55C, #f7f7f7)",
              }
            }
          />
          <div
            className="s-skeleton__element"
            style={
              Object {
                "background": "linear-gradient(to right, #F6D55C, #f7f7f7)",
              }
            }
          />
          <div
            className="s-skeleton__element"
            style={
              Object {
                "background": "linear-gradient(to right, #F6D55C, #f7f7f7)",
              }
            }
          />
        </div>,
        <div
          className="s-skeleton"
        >
          <div
            className="s-skeleton__element"
            style={
              Object {
                "background": "linear-gradient(to right, #F6D55C, #f7f7f7)",
              }
            }
          />
          <div
            className="s-skeleton__element"
            style={
              Object {
                "background": "linear-gradient(to right, #F6D55C, #f7f7f7)",
              }
            }
          />
          <div
            className="s-skeleton__element"
            style={
              Object {
                "background": "linear-gradient(to right, #F6D55C, #f7f7f7)",
              }
            }
          />
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="s-skeleton__element"
              style={
                Object {
                  "background": "linear-gradient(to right, #F6D55C, #f7f7f7)",
                }
              }
            />,
            <div
              className="s-skeleton__element"
              style={
                Object {
                  "background": "linear-gradient(to right, #F6D55C, #f7f7f7)",
                }
              }
            />,
            <div
              className="s-skeleton__element"
              style={
                Object {
                  "background": "linear-gradient(to right, #F6D55C, #f7f7f7)",
                }
              }
            />,
          ],
          "className": "s-skeleton",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "className": "s-skeleton__element",
              "style": Object {
                "background": "linear-gradient(to right, #F6D55C, #f7f7f7)",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "className": "s-skeleton__element",
              "style": Object {
                "background": "linear-gradient(to right, #F6D55C, #f7f7f7)",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "className": "s-skeleton__element",
              "style": Object {
                "background": "linear-gradient(to right, #F6D55C, #f7f7f7)",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="s-skeleton__element"
              style={
                Object {
                  "background": "linear-gradient(to right, #F6D55C, #f7f7f7)",
                }
              }
            />,
            <div
              className="s-skeleton__element"
              style={
                Object {
                  "background": "linear-gradient(to right, #F6D55C, #f7f7f7)",
                }
              }
            />,
            <div
              className="s-skeleton__element"
              style={
                Object {
                  "background": "linear-gradient(to right, #F6D55C, #f7f7f7)",
                }
              }
            />,
          ],
          "className": "s-skeleton",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "className": "s-skeleton__element",
              "style": Object {
                "background": "linear-gradient(to right, #F6D55C, #f7f7f7)",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "className": "s-skeleton__element",
              "style": Object {
                "background": "linear-gradient(to right, #F6D55C, #f7f7f7)",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "className": "s-skeleton__element",
              "style": Object {
                "background": "linear-gradient(to right, #F6D55C, #f7f7f7)",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="s-skeleton"
          >
            <div
              className="s-skeleton__element"
              style={
                Object {
                  "background": "linear-gradient(to right, #F6D55C, #f7f7f7)",
                }
              }
            />
            <div
              className="s-skeleton__element"
              style={
                Object {
                  "background": "linear-gradient(to right, #F6D55C, #f7f7f7)",
                }
              }
            />
            <div
              className="s-skeleton__element"
              style={
                Object {
                  "background": "linear-gradient(to right, #F6D55C, #f7f7f7)",
                }
              }
            />
          </div>,
          <div
            className="s-skeleton"
          >
            <div
              className="s-skeleton__element"
              style={
                Object {
                  "background": "linear-gradient(to right, #F6D55C, #f7f7f7)",
                }
              }
            />
            <div
              className="s-skeleton__element"
              style={
                Object {
                  "background": "linear-gradient(to right, #F6D55C, #f7f7f7)",
                }
              }
            />
            <div
              className="s-skeleton__element"
              style={
                Object {
                  "background": "linear-gradient(to right, #F6D55C, #f7f7f7)",
                }
              }
            />
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="s-skeleton__element"
                style={
                  Object {
                    "background": "linear-gradient(to right, #F6D55C, #f7f7f7)",
                  }
                }
              />,
              <div
                className="s-skeleton__element"
                style={
                  Object {
                    "background": "linear-gradient(to right, #F6D55C, #f7f7f7)",
                  }
                }
              />,
              <div
                className="s-skeleton__element"
                style={
                  Object {
                    "background": "linear-gradient(to right, #F6D55C, #f7f7f7)",
                  }
                }
              />,
            ],
            "className": "s-skeleton",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "className": "s-skeleton__element",
                "style": Object {
                  "background": "linear-gradient(to right, #F6D55C, #f7f7f7)",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "className": "s-skeleton__element",
                "style": Object {
                  "background": "linear-gradient(to right, #F6D55C, #f7f7f7)",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "host",
              "props": Object {
                "className": "s-skeleton__element",
                "style": Object {
                  "background": "linear-gradient(to right, #F6D55C, #f7f7f7)",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="s-skeleton__element"
                style={
                  Object {
                    "background": "linear-gradient(to right, #F6D55C, #f7f7f7)",
                  }
                }
              />,
              <div
                className="s-skeleton__element"
                style={
                  Object {
                    "background": "linear-gradient(to right, #F6D55C, #f7f7f7)",
                  }
                }
              />,
              <div
                className="s-skeleton__element"
                style={
                  Object {
                    "background": "linear-gradient(to right, #F6D55C, #f7f7f7)",
                  }
                }
              />,
            ],
            "className": "s-skeleton",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "className": "s-skeleton__element",
                "style": Object {
                  "background": "linear-gradient(to right, #F6D55C, #f7f7f7)",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "className": "s-skeleton__element",
                "style": Object {
                  "background": "linear-gradient(to right, #F6D55C, #f7f7f7)",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "host",
              "props": Object {
                "className": "s-skeleton__element",
                "style": Object {
                  "background": "linear-gradient(to right, #F6D55C, #f7f7f7)",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
